src/main/java/wolox/training/services/BookService.java

```diff
@@ -11,28 +15,44 @@
 import wolox.training.models.Book;
 import wolox.training.models.User;
 import wolox.training.repositories.BookRepository;
+import wolox.training.utils.VariablesConstants;
 
 @Service
 public class BookService {
 
-    @ApiOperation(value = "Given the isbn of a book, search in DB or external service and return the book or an exception", response = User.class)
-    @ResponseStatus(HttpStatus.NOT_FOUND)
-    public Book foundBookForGet(String isbn, BookRepository bookRepository, OpenLibraryService openLibraryService){
-      Optional<Book> list = bookRepository.findByIsbn(isbn);
-      if (list.isEmpty()){
-        return openLibraryService.bookInfo(isbn);
-      }
-      return list.get();
+  @ApiOperation(value = "Given the isbn of a book, search in DB or external service and return the book or an exception", response = User.class)
+  @ResponseStatus(HttpStatus.NOT_FOUND)
+  public Book foundBookForGet(String isbn, BookRepository bookRepository, OpenLibraryService openLibraryService){
+    Optional<Book> list = bookRepository.findByIsbn(isbn);
+    if (list.isEmpty()){
+      return openLibraryService.bookInfo(isbn);
     }
+    return list.get();
+  }
 
-    @ApiOperation(value = "Given the filter for book, return the books or an exception", response = User.class)
-    @ResponseStatus(HttpStatus.NOT_FOUND)
-    public List<Book> getAllBooks(String editor, String author, String genre, String year, String image,
-        String title, String subtitle, String page, String isbn, BookRepository bookRepository) {
-      List<Book> list = bookRepository.getAllBook(editor, author, genre, year, image, title, subtitle, page, isbn);
-      if (list.isEmpty()){
-        new BookHttpErrors("Book Not Found").bookNotFound();
-      }
-      return list;
+  @ApiOperation(value = "Given the filter for book, return the books or an exception", response = User.class)
+  @ResponseStatus(HttpStatus.NOT_FOUND)
+  public Page<Book> getAllBooks(String editor, String author, String genre, String year, String image,
+      String title, String subtitle, String page, String isbn, String orderByField, String orderBy,
```

Comment:
Replace String orderByField, String orderBy, String offse and String limit variables with a Pageable object